package pb;

import "resource.proto";

option java_package = "com.google.protobuf.implementation";
option java_outer_classname = "CameraProtos";

message Camera {
    extend Resource {
        required Camera resource = 100;
    }

    enum MotionType {
       Default      = 0;
       HardwareGrid = 1;
       SoftwareGrid = 2;
       MotionWindow = 4;
       NoMotion     = 8;
    }

    enum Quality {
        Lowest  = 0;
        Low     = 1;
        Normal  = 2;
        High    = 3;
        Highest = 4;
        Preset  = 5;
    }

    message ScheduleTask {
        required int32   id              = 1;         
        required int32   sourceId        = 2;
        required int32   startTime       = 3;
        required int32   endTime         = 4;
        required bool    doRecordAudio   = 5; 
        required int32   recordType      = 6;
        required int32   dayOfWeek       = 7;
        required int32   beforeThreshold = 8;
        required int32   afterThreshold  = 9;
        required Quality streamQuality   = 10;
        required int32   fps             = 11;
    }

    optional bool         scheduleDisabled = 1;
    optional MotionType   motionType       = 2;
    optional string       region           = 3;
    optional string       mac              = 4;
    optional string       login            = 5;
    optional string       password         = 6;
    repeated ScheduleTask scheduleTask     = 7;
    optional bool         audioEnabled     = 8;
    optional string       physicalId       = 9;
    optional bool         manuallyAdded    = 10;
}
